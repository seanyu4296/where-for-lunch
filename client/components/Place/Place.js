import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import isEmpty from 'lodash/isEmpty';
import styles from './Place.css';

const Place = ({ place: { data } }) => {
  const { name, address, phone, categories, price, reviewCount, rating, id } =
    data || {};
  return (
    <div className={styles.root}>
      <div className={styles.name}>{name || 'Where for lunch?'}</div>
      {isEmpty(data) ? (
        <div className={styles.box} />
      ) : (
        <Link to={`/place/${id}`} className={styles.box}>
          <div>{address}</div>
          <div>{phone}</div>
          <div>{categories && categories.join(', ')}</div>
          <div>{price}</div>
          {rating && (
            <div className={styles.rating}>
              <div className={styles.ratingScore}>{reviewCount} reviews</div>
              <div className={styles.stars}>
                <div className={styles.emptyStars} />
                <div
                  className={styles.fullStars}
                  style={{ width: `${(rating / 5) * 100}%` }}
                />
              </div>
            </div>
          )}
        </Link>
      )}
    </div>
  );
};

Place.propTypes = {
  place: PropTypes.object,
};

export default Place;
