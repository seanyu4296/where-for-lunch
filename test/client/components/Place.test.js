import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Place from 'client/components/Place/Place';
import renderer from 'react-test-renderer';

test('With Enzyme, Place component not render rating section when no rating passed over', () => {
  const place = { hehe: 'haha' };
  const wrapper = shallow(
    <MemoryRouter keyLength={0}>
      <Place place={place} />
    </MemoryRouter>,
  );
  const p = wrapper.find('.rating');
  expect(p.length).toBe(0);
});

test('With Jest snapshot, Place component renders rating section when present', () => {
  const place = { data: { rating: 3.5, id: 'test' } };
  const placeComponent = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <Place place={place} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(placeComponent).toMatchSnapshot();
});
