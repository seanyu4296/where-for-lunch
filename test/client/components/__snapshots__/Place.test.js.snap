// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Place component renders rating section when present 1`] = `
<div
  className="root"
>
  <div
    className="name"
  >
    Where for lunch?
  </div>
  <div
    className="box"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
